src/frontend/pages/homepage/homepage.jsx

```diff
@@ -1,33 +1,112 @@
-import {BsSortDownAlt, BsImageFill, BsFillEmojiSmileFill, BsFillCalendarCheckFill} from "react-icons/bs";
-import {AiOutlineGif} from "react-icons/ai";
-import {BiPoll} from "react-icons/bi";
-import {MdLocationOn} from "react-icons/md";
+import { useState } from "react";
+import LeftNav from "frontend/components/leftNav.jsx";
+import RightNav from "frontend/components/rightNav.jsx";
+import { BsSortDownAlt, BsImageFill, BsFillEmojiSmileFill, BsFillCalendarCheckFill } from "react-icons/bs";
+import { AiOutlineGif } from "react-icons/ai";
+import { BiPoll } from "react-icons/bi";
+import { GiCancel } from "react-icons/gi";
+import { MdLocationOn } from "react-icons/md";
+import Posts from "frontend/components/posts.jsx";
+import { v4 as uuid } from "uuid";
+import { usePosts } from "frontend/context/postContext";
+import { useToast } from 'frontend/custom/useToast';
+import { useUser } from 'frontend/context/userContext';
+import date from 'date-and-time';
 
 const Homepage = () => {
+    const [content, setContent] = useState("");
+    const [image, setImage] = useState("");
+    const { postsDispatch } = usePosts();
+    const { showToast } = useToast();
+    const { loggedUserData } = useUser();
+
+    const selectImageFile = (e) => {
+        e.preventDefault()
+        setImage(e.target.files[0])
+    }
+
+    const uploadPost = (e) => {
+        e.preventDefault()
+        if (!content && !image) {
+            showToast("Nothing to post", 'warning');
+            return
+        }
+        const newPost = !image ? {
+            _id: uuid(),
+            content: content,
+            image: "",
+            username: "Admin",
+            accountName: "@admin",
+            profileImage: "https://picsum.photos/60/60",
+            createdAt: date.format(new Date(), 'HH:mm, YYYY/MM/DD'),
+            likes: {
+                likeCount: 0,
+                likedBy: [],
+            },
+            scale: {
+                scaleCount: 0,
+                scaledBy: []
+            },
+            comments: [],
+        } : {
+            _id: uuid(),
+            content: content,
+            image: URL.createObjectURL(image),
+            username: "Admin",
```

Comment:
No need to send all the fields in while making a post. The backend will take care of adding all attributes. You can simply send the two strings. One for post text and another for image.